### PR TITLE
Initial addition of delayed status message printing 

### LINKED ISSUE #13569: Timed Print
## Reason
We need an object that will print out a message after a certain amount of time... and print a `.` every once in a while if something is taking a long time.

## Design
My idea here is to make it act like a "scope guard" like so:

```C++
{
  TimedPrint tp("message", 5);
  // possibly long running thing
}
```

This would print `"message"` after 5 seconds... and every second after that would print a `.`.

## Impact
Better information for long running operations.

### PR DESCRIPTION
**Pull request by friedmud**
Adds the macro: `CONSOLE_TIMED_PRINT()` that works similarly to the `TIME_SECTION` macro.  It waits a moment then prints the status message if the process is taking too long.

closes #13569

### CODE REVIEW

File: framework/src/mesh/MooseMesh.C [hunk 1]
```diff
@@ -2140,7 +2163,8 @@ MooseMesh::effectiveSpatialDimension() const
 {
   const Real abs_zero = 1e-12;
 
-  // See if the mesh is completely containd in the z and y planes to calculate effective spatial dim
+  // See if the mesh is completely containd in the z and y planes to calculate effective spatial
+  // dim
```
Comment (permcody) [inline framework/src/mesh/MooseMesh.C]: Hehe - I know what happened here. You indented it and saved, then unindented it. This has happened to me many times.

**Review by permcody — APPROVED** [review]
Nice work! This should be really useful.

**Comment by permcody** [comment]
@idaholab/moose-team - Well I didn't turn on any additional testing on this PR and it failed min compilers and valgrind testing. The Valgrind issue is easy enough to fix. Does anyone have a min to take a look at the min-compiler failure? It's related to arg packs and lambda capture. I have to believe that there is a solution to this (even with the old compiler), cause arg packs are supports on the old compiler. I'd prefer not to have to move forward our min compiler on a whim if we can avoid it. We might have to revert in the short term if we can't find a solution.

**Comment by lindsayad** [comment]
https://stackoverflow.com/questions/22568642/parameter-packs-not-expanded-with-another-variadic-template-bug-with-gc

It seems to me like a min compiler upgrade is in order

**Comment by permcody** [comment]
Well dang... Even if we decide to upgrade the minimum, it'll take Jason some time to build a new package and test, etc., etc. We may have to revert this until that's done.

**Pull request merged**